sklearn/preprocessing/_polynomial.py

```diff
@@ -393,18 +404,56 @@ def transform(self, X):
                 )
             if self._min_degree <= 1 and self._max_degree > 0:
                 to_stack.append(X)
+
+            # Covnert current stack to valid indices/indptr types
+            for mat in to_stack:
+                mat.indices = mat.indices.astype(index_t, copy=False)
+                mat.indptr = mat.indptr.astype(index_t, copy=False)
```

Comment:
Is this really needed? `to_stack` only has the optional intercept column and `X` at this point.

Also we should not mutate the input matrix `X` silently. This can cause very weird to debug issues in case scikit-learn is used a multi-threaded program for instance.